Building on OpenBSD 6.1 fails: unrecognized command line option "-Wno-array-bounds"

I followed the [build instructions for OpenBSD](https://github.com/neovim/neovim/wiki/Building-Neovim):

```
doas pkg_add gmake cmake libtool unzip autoconf-2.69p2 automake-1.15p0
export AUTOCONF_VERSION=2.69
export AUTOMAKE_VERSION=1.15

mkdir .deps
cd .deps
cmake ../third-party/
gmake
cd ..
mkdir build
cd build
cmake ..
gmake
```
On the last step I get this error:
```
[  0%] Building C object src/nvim/CMakeFiles/nvim.dir/auto/msgpack_lua_c_bindings.generated.c.o
cc1: error: unrecognized command line option "-Wno-array-bounds"
gmake[2]: *** [src/nvim/CMakeFiles/nvim.dir/build.make:1187: src/nvim/CMakeFiles/nvim.dir/auto/msgpack_lua_c_bindings.generated.c.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:4441: src/nvim/CMakeFiles/nvim.dir/all] Error 2
gmake: *** [Makefile:150: all] Error 2
```
I have GNU make 4.2.1 and cmake 3.7.2.. Short research revealed `-Wno-array-bounds` was introduced with gcc 4.6. I have no idea how to install the newer version. Upgrading the systems default compiler is said to be a bad idea as it may be used during a system update (but that's mere 3rd party "knowledge" on my side).

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -254,7 +254,10 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
   if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.8.5")
     # Array-bounds testing is broken in some GCC versions before 4.8.5.
     # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=56273
-    add_definitions(-Wno-array-bounds)
+    check_c_compiler_flag(-Wno-array-bounds HAS_NO_ARRAY_BOUNDS_FLAG)
+    if(HAS_NO_ARRAY_BOUNDS_FLAG)
+      add_definitions(-Wno-array-bounds)
+    endif()
   endif()
 endif()
 

--- a/src/nvim/regexp.c
+++ b/src/nvim/regexp.c
@@ -3316,6 +3316,47 @@ bt_regexec_nl (
   return (int)r;
 }
 
+/// Wrapper around strchr which accounts for case-insensitive searches and
+/// non-ASCII characters.
+///
+/// This function is used a lot for simple searches, keep it fast!
+///
+/// @param  s  string to search
+/// @param  c  character to find in @a s
+///
+/// @return  NULL if no match, otherwise pointer to the position in @a s
+static inline char_u *cstrchr(const char_u *const s, const int c)
+  FUNC_ATTR_PURE FUNC_ATTR_WARN_UNUSED_RESULT FUNC_ATTR_NONNULL_ALL
+  FUNC_ATTR_ALWAYS_INLINE
+{
+  if (!rex.reg_ic) {
+    return vim_strchr(s, c);
+  }
+
+  // Use folded case for UTF-8, slow! For ASCII use libc strpbrk which is
+  // expected to be highly optimized.
+  if (c > 0x80) {
+    const int folded_c = utf_fold(c);
+    for (const char_u *p = s; *p != NUL; p += utfc_ptr2len(p)) {
+      if (utf_fold(utf_ptr2char(p)) == folded_c) {
+        return (char_u *)p;
+      }
+    }
+    return NULL;
+  }
+
+  int cc;
+  if (ASCII_ISUPPER(c)) {
+    cc = TOLOWER_ASC(c);
+  } else if (ASCII_ISLOWER(c)) {
+    cc = TOUPPER_ASC(c);
+  } else {
+    return vim_strchr(s, c);
+  }
+
+  char tofind[] = { (char)c, (char)cc, NUL };
+  return (char_u *)strpbrk((const char *)s, tofind);
+}
 
 /// Matches a regexp against multiple lines.
 /// "rmp->regprog" is a compiled regexp as returned by vim_regcomp().
@@ -6318,42 +6359,6 @@ static int cstrncmp(char_u *s1, char_u *s2, int *n)
   }
 
   return result;
-}
-
-/*
- * cstrchr: This function is used a lot for simple searches, keep it fast!
- */
-static inline char_u *cstrchr(const char_u *const s, const int c)
-  FUNC_ATTR_PURE FUNC_ATTR_WARN_UNUSED_RESULT FUNC_ATTR_NONNULL_ALL
-  FUNC_ATTR_ALWAYS_INLINE
-{
-  if (!rex.reg_ic) {
-    return vim_strchr(s, c);
-  }
-
-  // Use folded case for UTF-8, slow! For ASCII use libc strpbrk which is
-  // expected to be highly optimized.
-  if (c > 0x80) {
-    const int folded_c = utf_fold(c);
-    for (const char_u *p = s; *p != NUL; p += utfc_ptr2len(p)) {
-      if (utf_fold(utf_ptr2char(p)) == folded_c) {
-        return (char_u *)p;
-      }
-    }
-    return NULL;
-  }
-
-  int cc;
-  if (ASCII_ISUPPER(c)) {
-    cc = TOLOWER_ASC(c);
-  } else if (ASCII_ISLOWER(c)) {
-    cc = TOUPPER_ASC(c);
-  } else {
-    return vim_strchr(s, c);
-  }
-
-  char tofind[] = { (char)c, (char)cc, NUL };
-  return (char_u *)strpbrk((const char *)s, tofind);
 }
 
 /***************************************************************

